src/main/java/baza/trainee/controller/AdminController.java

```diff
@@ -0,0 +1,33 @@
+package baza.trainee.controller;
+
+import baza.trainee.service.AdminService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+
+        /**
+        * AdminController handles all the admin-specific functionalities.
+        */
+
+@RestController
+@RequestMapping("/admin")
+@PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
Why you are using Method Security if you set request matches in a config file? Do not do that.